src/NServiceBus.RavenDB/Subscriptions/RavenDbSubscriptionStorage.cs

```diff
@@ -26,18 +27,31 @@ protected override void Setup(FeatureConfigurationContext context)
                 persister.DisableAggressiveCaching = true;
             }
 
-            if (context.Settings.GetOrDefault<bool>(RavenDbSubscriptionSettingsExtensions.DisableSubscriptionVersioningKey))
+            if (context.Settings.TryGet(RavenDbSubscriptionSettingsExtensions.LegacySubscriptionVersioningKey, out bool useLegacy))
```

Comment:
If you want to avoid the nested `if` here, you could change this to `if (!context.Settings.TryGet()` { throw}`.